### ISSUE TITLE
3.0 Roadmap

### ISSUE THREAD

**Issue by prc5** [prc5/react-zoom-pan-pinch#322]
Hi community!

I know the level of maintenence has not been of the highest quality lately, but this time there is a chance for a big change. The library was affected by problems such as the war in Ukraine which shorten my free time and prevented me from contributing as much as I would like. We managed to establish the BetterTyped collective, we are a good group of devs who want to create open source and react-zoom-pan-pinch will find its place there.

Its development plan is as follows:

1. Resolve existing bugs and problems that you reported.
2. Lock the current functionality with tests - as many as possible to make sure it works fine for next, main releases
3. Prepare a new library structure, setup monorepo, better eslint, rewrite many of the components, significantly improve and speed them up
4. Separate the core logic into a subpackage so that it can be used without React
5. Equip the react package with additional features, such as new hooks and control options, as well as better customization possibilities
5. Create new and clear documentation with more examples

These are pretty general milestones, but as we work, we'll keep you updated on the progress.

The release of the library will be based on semantic release, thanks to which we will be able to deliver the required fixes and improvements faster.

Thanks for your support, I know I've let many of you down by not fixing bugs, but I'll try to make the library shine again. At the moment we are releasing fixes for the most pressing bugs like React 18 support.

I also want to say a big thanks to @janaagaard75 for the fork and support he offered in my absence!

**Comment by michaelcummings** [comment]
Hey there!

Thank you for the update. My team would love to setup a quick call with you to see if we can help support this project a bit if you can help us prioritize a few bugs we've found. We've noticed one specific bug that is a known issue from others that is a major sticking point for our usecase.

I'll reach out to you with Contact info in your Twitter DM's

**Comment by michaelcummings** [comment]
We'll maybe. You're DMs are closed. Can you give me the best way to reach you privately?

**Comment by prc5** [comment]
More and more bugs getting resolved! Thank you @michaelcummings for the cooperation!!

I think we are more-less finishing the 1. point, which is solving bugs/performance issues. Now it looks much better.
We also started preparation for the wide testing scenarios of our functionalities, ideal coverage would be 100%.

1. Almost ready - we have to go through some smaller reports.
2. Preparations - this will help us with bugfixing.
3. Starting soon.
4. Code spliting done, we will just need to move it into separate monorepo package.
5. Some of the features will start to be added soon, I already finished KeepScale component.
6. This will be much more prioritized now as it will be important to keep the versioning of the documentation. Ideally we will implement everything into docusaurus. Hopefully we will start soon 🚀
